polkadot/node/core/backing/src/tests/mod.rs

```diff
@@ -250,176 +252,364 @@ impl TestCandidateBuilder {
 	}
 }
 
-// Tests that the subsystem performs actions that are required on startup.
-async fn test_startup(virtual_overseer: &mut VirtualOverseer, test_state: &TestState) {
-	// Start work on some new parent.
-	virtual_overseer
-		.send(FromOrchestra::Signal(OverseerSignal::ActiveLeaves(ActiveLeavesUpdate::start_work(
-			new_leaf(test_state.relay_parent, 1),
-		))))
-		.await;
-
+async fn assert_validation_requests(
+	virtual_overseer: &mut VirtualOverseer,
+	validation_code: ValidationCode,
+) {
 	assert_matches!(
 		virtual_overseer.recv().await,
 		AllMessages::RuntimeApi(
-			RuntimeApiMessage::Request(parent, RuntimeApiRequest::AsyncBackingParams(tx))
-		) if parent == test_state.relay_parent => {
-			tx.send(Err(ASYNC_BACKING_DISABLED_ERROR)).unwrap();
+			RuntimeApiMessage::Request(_, RuntimeApiRequest::ValidationCodeByHash(hash, tx))
+		) if hash == validation_code.hash() => {
+			tx.send(Ok(Some(validation_code))).unwrap();
 		}
 	);
 
-	// Check that subsystem job issues a request for the session index for child.
 	assert_matches!(
 		virtual_overseer.recv().await,
 		AllMessages::RuntimeApi(
-			RuntimeApiMessage::Request(parent, RuntimeApiRequest::SessionIndexForChild(tx))
-		) if parent == test_state.relay_parent => {
-			tx.send(Ok(test_state.signing_context.session_index)).unwrap();
+			RuntimeApiMessage::Request(_, RuntimeApiRequest::SessionIndexForChild(tx))
+		) => {
+			tx.send(Ok(1u32.into())).unwrap();
 		}
 	);
 
-	// Check that subsystem job issues a request for a validator set.
 	assert_matches!(
 		virtual_overseer.recv().await,
 		AllMessages::RuntimeApi(
-			RuntimeApiMessage::Request(parent, RuntimeApiRequest::Validators(tx))
-		) if parent == test_state.relay_parent => {
-			tx.send(Ok(test_state.validator_public.clone())).unwrap();
+			RuntimeApiMessage::Request(_, RuntimeApiRequest::SessionExecutorParams(sess_idx, tx))
+		) if sess_idx == 1 => {
+			tx.send(Ok(Some(ExecutorParams::default()))).unwrap();
 		}
 	);
 
-	// Check that subsystem job issues a request for the validator groups.
 	assert_matches!(
 		virtual_overseer.recv().await,
 		AllMessages::RuntimeApi(
-			RuntimeApiMessage::Request(parent, RuntimeApiRequest::ValidatorGroups(tx))
-		) if parent == test_state.relay_parent => {
-			tx.send(Ok(test_state.validator_groups.clone())).unwrap();
+			RuntimeApiMessage::Request(_, RuntimeApiRequest::NodeFeatures(sess_idx, tx))
+		) if sess_idx == 1 => {
+			tx.send(Ok(NodeFeatures::EMPTY)).unwrap();
 		}
 	);
+}
 
-	// Check that subsystem job issues a request for the availability cores.
+async fn assert_validate_from_exhaustive(
```

Comment:
nit: It might be more convenient to wrap assertions in macros instead of functions. If nothing has changed recently, failed tests with functions point to the line in the function itself rather than the specific line in the test.